R/DataUpgrade.R

```diff
@@ -0,0 +1,55 @@
+
+#' Replace treatment IDs with treatment names.
+#' 
+#' @param old_data Data frame with old data format where treatments are specified as IDs
+#' @param treatment_df Data frame with 'Number' column defining treatment IDs and 'Label' column defining treatment names
+#' @return Data frame in the new format with treatment names
+ReplaceTreatments <- function(old_data, treatment_df) {
+  upgraded <- CleanData(old_data)
+  if ('T' %in% colnames(upgraded)) {
+    # Long format
+    upgraded$T <- treatment_df$Label[match(upgraded$T, treatment_df$Number)]
+  } else {
+    # Wide format
+    for (col in paste0('T.', seq(6))) {
+      if (!(col %in% colnames(upgraded))) {
+        break
+      }
+      upgraded[[col]] <- treatment_df$Label[match(upgraded[[col]], treatment_df$Number)]
+    }
+  }
+  return(upgraded)
+}
+
+#' Create data frame containing treatment IDs and names.
+#' 
+#' @param treatment_names_string String containing treatment names, separated by commas
```

Comment:
I intentionally made this a simple list because the whole columns thing was the pain point. This will require a small amount of reformatting for people, but it'll be much more robust, so I think it's worth the effort. It's very difficult to mess up a comma separated list, and I've done some data tidying anyway for any minor errors.